pocs/scheduler/constraint.py

```diff
@@ -159,3 +165,160 @@ def get_score(self, time, observer, observation, **kwargs):
+        thresh = threshold_otsu(image)
+        binary = image > thresh
+
+        # Compute the Canny filter
+        edges1 = feature.canny(binary, low_threshold=0.1, high_threshold=0.5)
+
+        # Turn into array
+        np.set_printoptions(threshold=np.nan)
+        print(edges1.astype(np.float))
+
+    def get_config_coords(self):
+        """
+        Retrieves the coordinate list from the config file and validates it
+        If valid sets up a value for obstruction_points, otherwise leaves it empty
+        """
+
+        from pocs.tests.test_horizon_limits import obstruction_points_valid
+
+        self.obstruction_points = self.config['location']['horizon_constraint']
+
+        print("get_config_coords", "obstruction_points", self.obstruction_points)
+
+        if not obstruction_points_valid(self.obstruction_points):
+            self.obstruction_points = []
+
+    def enter_coords(self):
+        """
+        Enters a coordinate list from the user and validates it
+        If valid sets up a value for obstruction_points, otherwise leaves it empty
+        """
+
+        from pocs.tests.test_horizon_limits import obstruction_points_valid
+        print("Enter a list of azimuth elevation tuples with increasing azimuths.")
+        print("For example (10,10), (20,20), (340,70), (350,80)")
+
+        self.obstruction_points = input()
+        if not obstruction_points_valid(self.obstruction_points):
+            self.obstruction_points = []
+
+    def interpolate(self, A, B, az):
+        """
+        Determine the line equation between two points to return the elevation for a given azimuth
+
+        Args:
+            A (tuple): obstruction point A
+            B (tuple): obstruction point B
+            az (float or int): the target azimuth
+        """
+
+        # Input validation assertions.
+        assert len(A) == 2
+        assert len(B) == 2
+        assert type(az) == float or type(az) == int
+        assert type(A[0]) == float or type(A[0]) == int
+        assert type(A[1]) == float or type(A[1]) == int
+        assert type(B[0]) == float or type(B[0]) == int
+        assert type(B[1]) == float or type(B[1]) == int
+        assert az >= A[0]
+        assert az <= B[0]
+        assert az < 90
+
+        x1 = A[0]
+        y1 = A[1]
+        x2 = B[0]
+        y2 = B[1]
+
+        if x2 == x1:  # Vertical Line
+            el = max(y1, y2)
+        else:
+            m = ((y2 - y1) / (x2 - x1))
+            b = y1 - m * x1
+            el = m * az + b
+
+        assert el < 90
+
+        return el
+
+    def determine_el(self, az):
+        """
+        # Determine if the target altitude is above or below the determined minimum elevation for that azimuth
+
+        Args:
+            az (float or int): the target azimuth
+        """
+
+        el = 0
+        prior_point = self.obstruction_points[0]
+        i = 1
+        found = False
+        while(i < len(self.obstruction_points) and found is False):
```

Comment:
`while` loops are generally used when you have continuous looping until some breaking condition. Here you have a set number of points you want to loop over so might as well loop them directly, something like:
```python
for i, point in enumerate(self.obstruction_points):
    if found:
        break
    ....
```
Then just use `point` instead of `next_point` assignment below.